components/base/agilex/limo_base.go

```diff
@@ -330,12 +330,19 @@ func (lb *limoBase) setMotionCommand(linearVel float64,
 	return nil
 }
 
+// positive angleDeg spins base left. degsPerSec is a positive angular velocity.
 func (lb *limoBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
 	lb.logger.Debugf("Spin(%f, %f)", angleDeg, degsPerSec)
 	secsToRun := math.Abs(angleDeg / degsPerSec)
 	var err error
 	if lb.driveMode == DIFFERENTIAL.String() || lb.driveMode == OMNI.String() {
-		err = lb.SetVelocity(ctx, r3.Vector{}, r3.Vector{Z: degsPerSec}, extra)
+		sign := 0.0
+		if angleDeg < 0 {
```

Comment:
not sure if there's a cleaner way to do this sign check

Reply:
I think this function might be what we are looking for https://go.dev/src/math/copysign.go

Reply:
We usually do
```go
dir := 1.0
if math.Signbit(angleDeg) {
dir = -1.0
}
``` 

or some version thereof